services/task_store/service.go

```diff
@@ -706,56 +762,95 @@ func (ts *Service) IsEnabled(name string) (e bool) {
 	return
 }
 
-func (ts *Service) GetTaskSummaryInfo(tasks []string) ([]TaskSummaryInfo, error) {
-	taskInfos := make([]TaskSummaryInfo, 0)
+// Returns all taskInfo of task name that matches the predicate
+func (ts *Service) FindTasks(predicate func(string) (bool, error)) ([]*rawTask, error) {
+	rawTasks := make([]*rawTask, 0)
 
 	err := ts.db.View(func(tx *bolt.Tx) error {
 		tb := tx.Bucket([]byte(tasksBucket))
 		if tb == nil {
 			return nil
 		}
-		eb := tx.Bucket([]byte(enabledBucket))
-		// Grab task info
-		f := func(k, v []byte) error {
-			t, err := ts.LoadRaw(string(k))
+
+		return tb.ForEach(func(k, v []byte) error {
+			taskName := string(k)
+			isMatched, err := predicate(taskName)
+			if err != nil {
+				return err
+			}
+			if !isMatched {
+				return nil
+			}
+
+			// Grab task info
+			t, err := ts.LoadRaw(taskName)
 			if err != nil {
 				return fmt.Errorf("found invalid task in db. name: %s, err: %s", string(k), err)
 			}
 
-			enabled := eb != nil && eb.Get(k) != nil
+			rawTasks = append(rawTasks, t)
+			return nil
+		})
+
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return rawTasks, nil
+}
+
+func (ts *Service) GetTaskSummaryInfo(tasks []string) ([]TaskSummaryInfo, error) {
+	var isEmptyTasks = len(tasks) == 0
+
+	rawTasks, err := ts.FindTasks(func(taskName string) (bool, error) {
+		if isEmptyTasks {
+			return true, nil
+		}
+
+		// This code is not the optimal code ever, we could use map here
+		// but I am assuming the asked tasks count will be small
+		for _, askedTaskName := range tasks {
+			if askedTaskName == taskName {
+				return true, nil
+			}
+		}
```

Comment:
Let's make sure we are on the same idea, you are suggesting that instead of using FindTasks and looking at the bucket we will do something like this:
```
for taskName := range tasks {
   raw:=ts.LoadRaw(taskName)
  // convert to TaskSummaryInfo, and IsEnabled, and ExecutionStatus
}
```
If you are suggesting doing this approach, we have the case where the tasks are empty and we want to fetch all tasks - "/tasks",  so we are using FindTasks to find the matching tasks.